### DISCUSSION TITLE
Is it possible to merge live and videos in the subscription feed.

### DISCUSSION THREAD

**Discussion by SentientBeingOnAQuest** [FreeTubeApp/FreeTube#4225] [Q&A] [answered]
I understand that it's handy to be able to filter out shorts and community posts from the default sub feed, but I'm missing so much content because I forget to check my live feed. 
Is there any way to include lives in the default videos subscription feed?

**Comment by absidue** [comment, marked as answer]
Having separate feeds was a deliberate design decision to avoid getting ratelimited by YouTube, so no unfortunately combining them would result in you having an even worse experience.

Having separate tabs adds enough delay between the request spam, that YouTube's systems don't get too mad.

Those tabs aren't filters, we have to make separate requests to all 4 channel tabs, for every channel you are subscribed to, that is a lot of requests to YouTube, in a very short space of time. By having separate tabs, we can delay fetching all the channel tabs for that tab until you click on it.

**Reply by SentientBeingOnAQuest** [reply]
That's a good point that I hadn't considered. 

Would limiting the amount of videos that are fetched by shortening the feed solve this problem? 
So perhaps give users the option to have the tabs combined at the expense of having a shorter feed in which you'd have to re-request older videos via a "show more" button at the bottom of the page?

Or maybe have different types of videos fetched through multiple different Invidious APIs so that Youtube's algorithm doesn't detect it as a single request?

**Reply by absidue** [reply]
It's not about how many videos FreeTube requests, the app already only requests one page per channel tab per channel. The problem is that there is no way around making all those requests if you want a fully functioning subscription feed.

I think you might be a bit confused at how FreeTube works, you seem to be imagining that there is one endpoint on YouTube's end that we use to get subscription videos, yes YouTube has that endpoint but we can't use it because you have to be logged into a Google account, with subscriptions in said Google account. As FreeTube is a privacy friendly app, there are no Google accounts involved. Instead FreeTube has to build the feed itself by getting the content from the videos, live, shorts and community tabs separately for every channel you are subscribed to. That requires making a lot of web requests, which is why FreeTube has the subscriptions page split up into four tabs, that way it can delay making the requests for the live tab until you select it.

The load more button doesn't do any new requests, it just shows more of the already fetched data (prevents performance problems caused by immediately showing everything).

Just to give you a picture, let's say for example you have 50 subscriptions (we are fully aware that there are people with hundreds of subscriptions):

1. You click on the videos tab on the subscriptions page: FreeTube requests the videos channel tab for every one of your subscriptions, that's 50 requests
2. You click on the Live tab on the subscriptions page: FreeTube requests the live channel tab for each of your subscriptions, that's 50 requests
3. You click on the shorts subscriptions tab: FreeTube requests the RSS feeds of the shorts channel tab for each of your subscriptions, as the channel tab itself doesn't show any published dates, which are quite necessary when you need to sort stuff by published date. The RSS feeds don't have any ratelimits or contribute to other ones, so the 50 requests here are fine

So you are already at 100 requests, add a few more subscriptions and you'll hit the 125 threshold, at which point FreeTube forces RSS to stop YouTube ratelimiting. The RSS feeds don't contain the durations of videos or the live and upcoming status of live streams, so  upcoming and live videos will not only not be labelled but they will also be in the wrong order in your live feed.

So yes as you can see, we could combine it but the user experience would be significantly worse.

**Reply by anyparktos** [reply]
Hi! Thank you for your detailed explanation. Could you please clarify if the threshold is 125 requests _per minute_? Also do you think it's a good idea to implement a feature where the user could select which channels should update via rss and which via yt feed in order to limit the requests? Because out of the 100s of my subscriptions only a handful go live regularly so it would be nice to have some extra data on them while still get the occasional live videos from the others via rss.

**Reply by absidue** [reply]
Currently that limit is based on the number of subscriptions you have, as we fire off all the requests for a given tab at the same time, but if we combined the feeds we would have to do it based on number of subscriptions x the number of requests per channel. (no other place in FreeTube does that many requests in one go)

Doing per channel selection for RSS or not, is a very niche use case, I think most people would prefer the separate tabs without having to lose information.

If you have more than 125 subscriptions we hide the community tab for that profile automatically, as YouTube doesn't provide any RSS feeds for that.

**Reply by VictorVow** [reply]
I'd like the ability to click "All" and have it load all, with a progress bar and text e.g. "Loading Videos..." and a few seconds later "Loading Shorts...", etc to save me clicking through each, and being able to eventually have the videos all on one page sorted by the date+time they were published. 

Maybe with an added tooltip of why it works this way, but there's still great utility in having this type of display available.

**Reply by alex-milanov** [reply]
why not allowing the user to choose to have a delayed list (as if they will be clicking consecutively on live and shorts)
So we get the videos and after 5s we get live and after 5s we get shorts.
And with some caching on the user end we'll just add the latest ones. (I think this is how NewPipe is refreshing)
The default would still be 3 tabs, but possibly in the general settings we could switch between the 2 modes (+ maybe specifying which video types to be included or prioritized), with the warning that the refresh times will take longer. And still we'll get the normal videos with the same speed so we'll have gradual refresh rather than a blank screen with "please wait".
What do you think?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Like @absidue chose for this design in the name of having a good UX. We will not be implementing something like this. Therefore im closing this discussion.

**Reply by alex-milanov** [reply]
Imo. this is precisely bad UX decision motivated by technical restriction or perceived technical restriction. 

As I mentioned different channels focus on different types of videos primarily, live, shorts or normal. 

I have no idea if they've released sth. unless I click through the tabs every time.
